test/extended/util/client.go

```diff
@@ -143,6 +143,32 @@ func NewCLIWithoutNamespace(project string) *CLI {
 	return cli
 }
 
+// NewHypershiftManagementCLI returns a CLI that interacts with the Hypershift management cluster.
+// Contrary to a normal CLI it does not perform any cleanup and it must not be used for any mutating
+// operations. Also contrary to a normal CLI it must be constructed inside an `It` block. This is
+// because retrieval of hypershift management cluster config can fail, but assertions are only
+// allowed inside an It block. AfterEach and BeforeEach are not allowed there though.
+func NewHypershiftManagementCLI(project string) *CLI {
```

Comment:
/assign @bparees 

Sorry I have to bother you again for the changes under test/extended/util (first commit).

I am not particularly happy with this behaving differently than a normal cli with the lack of cleanup, but its a chicken-egg problem, we only know if we need it after checking for the controlplane toplogy, in order to check we have to be in an It block, once we are in an it block we can not register the cleanup anymore. Ginkgo can be pretty frustrating to work with.

As we don't assume tests manipulate anything in the management cluster this seems okayish to me and I didn't manage to come up with a better alternative. Very open to suggestions though if you have any.

Reply:
thanks for explaining this and including it in the comment.

presumably there is nothing to actually clean up anyway because this doesn't create a new project?

Reply:
Yeah there is nothing to cleanup, hence it's okayish